collect_app/src/main/java/org/odk/collect/android/application/initialization/upgrade/AppUpgrader.kt

```diff
@@ -16,6 +16,16 @@ class AppUpgrader(private val metaSettings: Settings, private val upgrades: List
             }
         }
     }
+
+    fun install() {
```

Comment:
I'm not sure if I understand the purpose of keeping it... if you keep that and run in v2021.3.3 that is going to set flags for all migrators to true (do not perform in the future). Maybe not in the first run when you upgrade from an older version but then if you kill the app and open again it will happen.

Let's say we have a user who uses v2021.3.3 so all those flags are set to true on his device.
Then in v2022.1 you decide to rename one of our preferences (`ExistingSettingsMigrator` is responsible for that).
Our user upgrades form v2021.3.3 to v2022.1 but his `ExistingSettingsMigrator` is not trigger because the flag of this migrator was set to true.

Am i missing something?

Reply:
The thing is that we have two types of migrators:
- final migartors
- editable migrators

`ExistingProjectMigrator` is final if it was performed once we shouldn't need it anymore so setting it's flag to true is fine.
`ExistingSettingsMigrator` I would say it's not final we might need it every time we decide to update some prefs
`FormUpdatesUpgrade` here I'm not sure I would need to dig depper.

so you can't disable all of them unless I'm missing something.

Reply:
`Upgrades` can only be skipped if they have a non `null` `key()` (you can see in `AppUpgrader` or its tests). This lets us run `ExistingSettingsMigrator` and `FormUpdatesUpgrade` every upgrade and only run `ExistingProjectMigrator` once. Calling `install()` will not cause the first two to be skipped as they return `null` from their `key()` implementations so can't be marked as already having been run.

Reply:
Ok now I see but the way it woks (with those keys null or nonnull) is confusing...
So this will be run only for `ExistingProjectMigrator` so why is it important if we do the check inside `ExistingProjectMigrator` anyway?

Reply:
Ok for this PR we can just take out the `install()` call as I think it's confusing things. I do want this to go in `master` but we can discuss that separately as improvements to the way upgrades work rather than just a fix for this specific issue.